test/e2e/framework/service/jig.go

```diff
@@ -82,6 +111,61 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocatePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func (s *staticPortRange) allocatePort(port int) bool {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	if port < 0 || port > s.length || s.reservedPorts.Has(port) {
```

Comment:
I think it's confusing to have `s.reservedPorts` store offsets within the range, rather than the actual ports. I'd leave `port` as-is and just check `port < s.baseport || port > s. baseport + s.length` here (or store `.lastport` instead of `.length`, though it looks like that would be less convenient below).